[OSDOCS-16990] AWS IPI/Edge/Outposts: replace callouts and add abstracts (#118245)
openshift/openshift-docs: modules/installation-aws-upload-custom-rhcos-ami.adoc

:_mod-docs-content-type: PROCEDURE
[id="installation-aws-upload-custom-rhcos-ami_{context}"]
= Uploading a custom {op-system} AMI in {aws-first}

[role="_abstract"]
If you are deploying to a custom {aws-short} region, you must
upload a custom {op-system-first} Amazon Machine Image (AMI) that belongs to
that region.

.Prerequisites

* You configured an {aws-short} account.
* You created an Amazon S3 bucket with the required IAM
link:https://docs.aws.amazon.com/vm-import/latest/userguide/vmie_prereqs.html#vmimport-role[service role].
* You uploaded your {op-system} VMDK file to Amazon S3.
ifdef::openshift-enterprise,openshift-webscale[]
The {op-system} VMDK file must be the highest version that is less than or equal to the {product-title} version you are installing.
endif::[]
* You downloaded the {aws-short} CLI and installed it on your computer. See
link:https://docs.aws.amazon.com/cli/latest/userguide/install-bundle.html[Install the {aws-short} CLI Using the Bundled Installer].

.Procedure

. Export your {aws-short} profile as an environment variable:
+
[source,terminal]
----
$ export AWS_PROFILE=<aws_profile>
----
+

> The {aws-short} profile name that holds your {aws-short} credentials, such as `govcloud` or `beijingadmin`.

Reviewer: ```suggestion
Replace `<aws_profile>` with the profile name that holds your {aws-short} credentials, such as `govcloud` or `beijingadmin`.
```

But since Silke is on leave, I will quickly address in a follow up PR if no other glaring issues are found

. Export the region to associate with your custom AMI as an environment
variable:
+
[source,terminal]
----
$ export AWS_DEFAULT_REGION=<aws_region>
----
+

> The {aws-short} region, such as `us-gov-east-1` or `cn-north-1`.

Reviewer: ```suggestion
Replace `<aws_region>` with the {aws-short} region, such as `us-gov-east-1` or `cn-north-1`.
```

. Export the version of {op-system} you uploaded to Amazon S3 as an environment variable by running the following command:
+
[source,terminal]
----
$ export RHCOS_VERSION=<version>
----
+

> The {op-system} VMDK version, such as `{product-version}.0`.

Reviewer: ```suggestion
Replace `<version>` with the {op-system} VMDK version, such as `{product-version}.0`.
```

+
[source,terminal]
----
$ export AWS_PROFILE=<aws_profile>
----
+

> The AWS profile name that holds your AWS credentials, like `govcloud` or `beijingadmin`.

Reviewer 2: ```suggestion
The AWS profile name that holds your AWS credentials, such as `govcloud` or `beijingadmin`.
```
you're here, may as well tweak up the style compliance

> . Export the region to associate with your custom AMI as an environment

Reviewer 2: ```suggestion
. Export the region to associate with your custom AMI as an environment variable by running the following command:
```
we're missing accessibility on the commands, and it looks like there is a hard return in the content that needs removed; i would do a quick check for both cases throughout

Reviewer: Still missing "by running the following command" in a few instances, but Silke is on leave and was helping out my CQA work here, so I will just address that in my own PRs.

> variable:

Reviewer 2: ```suggestion
```

Author: I think "variable" belongs here but added "by running the following command"

+
[source,terminal]
----
$ export AWS_DEFAULT_REGION=<aws_region>
----
+

> The AWS region, like `us-gov-east-1` or `cn-north-1`.

Reviewer 2: ```suggestion
The AWS region, such as`us-gov-east-1` or `cn-north-1`.
```

> . Export the version of {op-system} you uploaded to Amazon S3 as an environment

Reviewer 2: same here

> variable:

Reviewer 2: ```suggestion
```

Author: I think "variable" belongs here but added "by running the following command"

+
[source,terminal]
----
$ export RHCOS_VERSION=<version>
----
+

> The {op-system} VMDK version, like `{product-version}.0`.

Reviewer 2: ```suggestion
The {op-system} VMDK version, such as `{product-version}.0`.
```
that doesn't even look correct; could you make a jira to check the .0? e.g, would this work with z-streams? a Q for the backlog

+
[source,terminal]
----
$ aws ec2 import-snapshot --region ${AWS_DEFAULT_REGION}

>      --description "<description>"

Reviewer: Missed this in my first review pass, but also these slashes need to be kept as they are syntax for the command itself and not formatting for the callouts. Will add this back in my follow up PR

     --disk-container "file://<file_path>/containers.json"
----
+
where:

`--description "<description>"`:: Specifies the description of your {op-system} disk being imported, like `rhcos-${RHCOS_VERSION}-x86_64-aws.x86_64`.

> `--disk-container "file://<file_path>/containers.json"`:: Specifies the file path to the JSON file describing your {op-system} disk. The JSON file should contain your Amazon S3 bucket name and key.

Reviewer: I'd list either the flags or the replaceable values, but not both:

```suggestion
`--description`:: Specifies the description of your {op-system} disk being imported, like `rhcos-${RHCOS_VERSION}-x86_64-aws.x86_64`.
`--disk-container"`:: Specifies the file path to the JSON file describing your {op-system} disk. The JSON file should contain your Amazon S3 bucket name and key.
```

----
$ aws ec2 register-image \
   --region ${AWS_DEFAULT_REGION} \
   --architecture x86_64 \
   --description "rhcos-${RHCOS_VERSION}-x86_64-aws.x86_64" \
   --ena-support \
   --name "rhcos-${RHCOS_VERSION}-x86_64-aws.x86_64" \
   --virtualization-type hvm \
   --root-device-name '/dev/xvda' \
   --block-device-mappings 'DeviceName=/dev/xvda,Ebs={DeleteOnTermination=true,SnapshotId=<snapshot_ID>}'
----
+
where:

> `--architecture`:: Specifies the {op-system} VMDK architecture type, like `x86_64`,

Reviewer 2: ```suggestion
`--architecture`:: Specifies the {op-system} VMDK architecture type, such as `x86_64`,
```

ifndef::openshift-origin[]
`aarch64`,
endif::openshift-origin[]
`s390x`, or `ppc64le`.
`--description`:: Specifies the `Description` from the imported snapshot.
`--name`:: Specifies the name of the {op-system} AMI.
`--block-device-mappings`:: Specifies the `SnapshotID` from the imported snapshot.

> To learn more about these APIs, see the AWS documentation for

Reviewer 2: not compliant; this will get stripped out during conversion because the link is not connected to the step

Author: Moving this to the assembly as additional resources.
